ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1030,13 +1037,13 @@ public void write()
           hardDeleter.preLogFlush();
 
           // flush the log to ensure everything till the fileEndPointerBeforeFlush is flushed
-          log.flush();
+          logSegment.flush();
```

Comment:
Yes - the changes in the index in this patch are done assuming one single segment. The next patch (that has index changes) will flush the `Log`.